### PR TITLE
fix interplanetary trade

### PR DESCRIPTION
**Pull request by vsrisuknimit**
Planetary Trade (card) should not count Venus tag or Moon tag by default.

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const MILESTONE_COST: number = 8;
 export const MAX_MILESTONES = 3;
 export const AWARD_COSTS: Array<number> = [8, 14, 20];
 export const MAX_AWARDS = 3;
+export const DEFAULT_TAG_COUNT = 10;
```
Comment (kberg) [inline src/constants.ts:7]: This only applies to the one card and it's ready. Seems like it should go there and not be a global constant. Are there other uses for this? If not, please hang it off the card.

File: src/cards/promo/InterplanetaryTrade.ts [hunk 2]
```diff
@@ -30,8 +31,10 @@ export class InterplanetaryTrade extends Card implements IProjectCard {
   public play(player: Player) {
     // This card tag is counting as well
     const availableTags = player.getDistinctTagCount(true, Tags.SPACE);
-    // Only count wildcards up to the max amount of tag types existing (minus events and wildcards)
-    const existingTags = Object.keys(Tags).length - 2;
+    // There is only 10 tags in the base game. One more for Venus and one more for Moon.
+    let existingTags = DEFAULT_TAG_COUNT;
+    if (player.game.gameOptions.venusNextExtension) existingTags++;
```
Comment (nwai90) [inline src/cards/promo/InterplanetaryTrade.ts:36]: While the code works, it is currently placed in the wrong scope, and would be better moved into the `getDistinctTagCount` function in `Player.ts` to handle any existing or future scenarios that require similar logic. Ideally `player.getDistinctTagCount(true, Tags.SPACE)` should return the correct result by considering which expansions are in play, i.e.

Screenshot 2021-07-24 at 11 05 01 PM

Comment (kberg) [reply src/cards/promo/InterplanetaryTrade.ts:36]: Good catch. That makes senes.

File: src/Player.ts [hunk 3]
```diff
@@ -831,7 +831,10 @@ export class Player implements ISerializable<SerializedPlayer> {
       }
     }
     if (countWild) {
-      return uniqueTags.size + wildcardCount;
+      let maxTagCount = constants.DEFAULT_TAG_COUNT;
```
Comment (kberg) [inline src/Player.ts:834]: This is the right place for this logic, but please, still remove the constant unless you expect it to be used elsewhere.

File: src/constants.ts (hunk 1 above)
Comment (bafolts) [reply src/constants.ts:7]: If we ever find another use case for this number we could create a constant at that time.

**Review by bafolts — CHANGES_REQUESTED** [review]
Can you change the `DEFAULT_TAG_COUNT` from a constant to a local variable?

**Review by nwai90 — APPROVED** [review]

**Review by bafolts — APPROVED** [review]

**Pull request merged**